### DISCUSSION TITLE
Why there is a line-break after ul>li elements?

### DISCUSSION THREAD

**Discussion by joisun** [markedjs/marked#3552] [Q&A] [answered]
The margin of ul/ol>li (and all other elements) doesn't looks as expected. 
 
![image](https://github.com/user-attachments/assets/196e94cf-101b-4d07-9079-e963d4c5da43)

while check the css of ul/ol and li , they are looks right:

**ul/ol:**
![image](https://github.com/user-attachments/assets/2ed73b7d-55ef-4e9d-9b18-a6bc874c9034)

**li:**
![image](https://github.com/user-attachments/assets/f99b6915-2025-426a-864b-6d9ec303343c)

But when I select the text, there seems a line-break after each li elements and all other elements:

![image](https://github.com/user-attachments/assets/71b3ce2d-8849-4584-860f-5045024ae64e)

How should I solve this problem? And what i did was just simple:

```js
marked.parse(#the markdown content#)
```
and the content is:
```markdown
 | Displayed on screen 🖥️
 | /dɪˈspleɪbt ɒn skriːn/
 | verb

**Definition:** To show or make something visible on a computer, television, or other electronic device's screen.

**Example sentences:**
1. The presentation was displayed on the big screen for everyone to see. 📊🖥️
2. She clicked on the video link, and it started displaying on her phone screen. 📱🎥
3. The website has been updated, and the new design is now displayed on screen. 💻🎨
```

Thanks a lot :)

**Comment by UziTech** [comment]
My guess is that it is your css that adds the margin. marked does not provide css. It looks fine in the [demo](https://marked.js.org/demo/?text=%20%7C%20Displayed%20on%20screen%20%F0%9F%96%A5%EF%B8%8F%0A%20%7C%20%2Fd%C9%AA%CB%88sple%C9%AAbt%20%C9%92n%20skri%CB%90n%2F%0A%20%7C%20verb%0A%0A**Definition%3A**%20To%20show%20or%20make%20something%20visible%20on%20a%20computer%2C%20television%2C%20or%20other%20electronic%20device%27s%20screen.%0A%0A**Example%20sentences%3A**%0A1.%20The%20presentation%20was%20displayed%20on%20the%20big%20screen%20for%20everyone%20to%20see.%20%F0%9F%93%8A%F0%9F%96%A5%EF%B8%8F%0A2.%20She%20clicked%20on%20the%20video%20link%2C%20and%20it%20started%20displaying%20on%20her%20phone%20screen.%20%F0%9F%93%B1%F0%9F%8E%A5%0A3.%20The%20website%20has%20been%20updated%2C%20and%20the%20new%20design%20is%20now%20displayed%20on%20screen.%20%F0%9F%92%BB%F0%9F%8E%A8&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%0A%7D&version=15.0.3)

**Reply by joisun** [reply]
> My guess is that it is your css that adds the margin. marked does not provide css. It looks fine in the [demo](https://marked.js.org/demo/?text=%20%7C%20Displayed%20on%20screen%20%F0%9F%96%A5%EF%B8%8F%0A%20%7C%20%2Fd%C9%AA%CB%88sple%C9%AAbt%20%C9%92n%20skri%CB%90n%2F%0A%20%7C%20verb%0A%0A**Definition%3A**%20To%20show%20or%20make%20something%20visible%20on%20a%20computer%2C%20television%2C%20or%20other%20electronic%20device%27s%20screen.%0A%0A**Example%20sentences%3A**%0A1.%20The%20presentation%20was%20displayed%20on%20the%20big%20screen%20for%20everyone%20to%20see.%20%F0%9F%93%8A%F0%9F%96%A5%EF%B8%8F%0A2.%20She%20clicked%20on%20the%20video%20link%2C%20and%20it%20started%20displaying%20on%20her%20phone%20screen.%20%F0%9F%93%B1%F0%9F%8E%A5%0A3.%20The%20website%20has%20been%20updated%2C%20and%20the%20new%20design%20is%20now%20displayed%20on%20screen.%20%F0%9F%92%BB%F0%9F%8E%A8&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%0A%7D&version=15.0.3)

Thanks for your help, but there is no custom css been set, I use tailwind typography to show the markdown,  I changed the `marked.parse` to `marked.parseInline` and things goes well.

![image](https://github.com/user-attachments/assets/79401350-0b7c-48bf-b2c2-c7be75853a81)

**Reply by UziTech** [reply]
By your css I don't mean custom css. If you are using tailwind it must be tailwind that is adding the margin.

`parseInline` won't create block markdown like lists. so the `1. ... 2. ... 3. ...` is text not an html list.

**Reply by joisun** [reply]
> By your css I don't mean custom css. If you are using tailwind it must be tailwind that is adding the margin.
> 
> `parseInline` won't create block markdown like lists. so the `1. ... 2. ... 3. ...` is text not an html list.

Thanks for the notice, and I just tried to create a minor reproduction [stackblitz](https://stackblitz.com/~/github.com/joisun/tmp-demo?file=src/App.tsx), Things works as expected。 

But In my local code base, If I keep everything the simplest, which the html structure is:
```tsx
           <div >
              <div
                className=""
                dangerouslySetInnerHTML={{ __html: htmlContent as string }} // 动态渲染的 HTML
              />
            </div>
```            

the page shows like:

![image](https://github.com/user-attachments/assets/f73a3dcf-f480-4578-9ba9-4e8e0cf75afa)


The point is I try to findout what css setting might effected this, I couldn't. 

Thanks for your time, This project is a browser extension, so It might not convenient to reproduce for other people, that's what makes me feel embarrassed, do you have any adivese for me to troubleshoot this problem?  appreciate : )

**Reply by UziTech** [reply]
I don't know much about tailwind. I would try removing css properties one by one to see which ones in tailwind add the margin. Or you could try adding custom css that overrides the margin.

**Reply by joisun** [reply, marked as answer]
I finally foundout the point, I add a style for the continer `whitespace-break-spaces` which has been inheritted.

Thanks a lot !
